chain/block.go

```diff
@@ -290,6 +291,23 @@ func (b *StatelessBlock) initializeBuilt(
 			b.containsWarp = true
 		}
 	}
+
+	// transaction hash generation
+	merkleItems := make([][]byte, 0, len(b.Txs)+len(b.results))
+	for _, tx := range b.Txs {
+		merkleItems = append(merkleItems, tx.Bytes())
+	}
+	for _, result := range b.results {
+		merkleItems = append(merkleItems, result.Output)
```

Comment:
Not sure what just storing hashes of outputs of results will provide? It seems like it should at least be associated with a tx...

Reply:
Yea, one way to do that is to concanate transaction with corresponding output then feed into merkle tree.